src/adapter/src/coord/introspection.rs

```diff
@@ -36,19 +37,80 @@ use crate::{
 
 /// Checks whether or not we should automatically run a query on the `mz_introspection`
 /// cluster, as opposed to whatever the current default cluster is.
-pub fn auto_run_on_introspection<'a, 's>(
+pub fn auto_run_on_introspection<'a, 's, 'p>(
     catalog: &'a Catalog,
-    session: &'s mut Session,
-    depends_on: impl IntoIterator<Item = GlobalId>,
-) -> Option<&'a Cluster> {
+    session: &'s Session,
+    plan: &'p Plan,
+) -> bool {
```

Comment:
as we mentioned offline, probably return `TargetCluster` here instead of `bool`

Reply:
Done.